Added notes explaining device connectivity status (heartbeat/vpn only) (#30)
balena-io/debugging-masterclass: README.md

includes an entry for doing this (search for 'Recovery data from a dead
device').

### 12. Device connectivity status

> When the device's network connectivity is fully operational the status displays as Online. However, these two conditions which make up this value may run into issues and result in Heartbeat Only or VPN Only.

Reviewer: ```suggestion
When the device's network connectivity is fully operational, the status displays as Online, which is comprised of healthy Heartbeat and VPN statuses. When one of the aforementioned conditions is abnormal, the status will display as Heartbeat Only or VPN Only.
```

#### 12.1 Heartbeat Only

> When a device is in this state this indicates that the device has internet connectivity and can poll the cloud for new updates to apply. However, if you try to perform any actions which do not change the target state or want access via SSH then you will encounter network issues because the device is only accessible from the VPN. Purging data, restarting/shutdown, etc, actions require the VPN because these requests are proxied from the dashboard API to the [Supervisor API](https://www.balena.io/docs/reference/supervisor/supervisor-api/) on device and so will fail with network issues as well.

Reviewer: ```suggestion
A device with a Heartbeat Only status has internet connectivity and can poll the cloud for new updates to apply. However, performing any actions which do not change the target state or attempting to access the device via SSH will error, as these actions only have effect with a VPN connection. Actions that do not change the target state include purging data, restarting, rebooting, or shutting down, and they all require the VPN because they are proxied from the dashboard API to the [Supervisor API](https://www.balena.io/docs/reference/supervisor/supervisor-api/) on device.
```

#### 12.2 VPN Only

> In this state the device is not able to apply any new changes made such as deploying now releases, applying service configuration values, or switching to localmode. It however is accessible via SSH or the web terminal! Any actions you perform such as reboot and container restart might work but most likely not. The reason for this is because the device loses its Heartbeat if it's not communicating with the API which is usually when the Supervisor service on the device is not running or crashing. Since we have VPN access we can SSH to the device and investigate further.

Reviewer: ```suggestion
A device with a VPN Only status is not able to apply any new changes made such as deploying new releases, applying service configuration values, or switching to local mode. However, it is accessible via SSH or the web terminal. Performing an action such as rebooting or restarting containers might work, but most likely will not. This is because the device loses its Heartbeat if it's not communicating with the API, which is usually when the Supervisor on the device is not running or crashing. Since we have VPN access, we can SSH into the device and investigate further.
```

#### 12.3 What do these states mean for my application ?

> A device can be Heartbeat or VPN Only and still have full internet access which means your applications may be deployed and continuing to run without interuption. In the case of VPN Only though, future updates will not be deployed until the Supervisor service is resolved.

Reviewer: ```suggestion
A device can be Heartbeat or VPN Only and still have full internet access, which means your applications may be deployed and continuing to run without interruption. However, in the case of VPN Only, future updates will not be deployed until the Supervisor on the device is fixed.
```

* Understand the Supervisor's role, including key concepts.
* Understand the balenaEngine's role, including key concepts.
* Be able to look at kernel logs, and determine some common faults.

> * Work with media issues, including dealing with full media, working with customer data and diagnosing corruption issues.

Reviewer: ```suggestion
* Work with media issues, including dealing with full media, working with customer data, and diagnosing corruption issues.
```
Oxford comma!
